dashboard/db/migrate/20170215183924_modify_script_level_properties.rb

```diff
@@ -0,0 +1,24 @@
+class ModifyScriptLevelProperties < ActiveRecord::Migration[5.0]
+  # ScriptLevel.properties is currently a hash representing variant levels. We'd
+  # like the property hash to be more generic, so we're moving the variants to
+  # exist under their own key within properties
+
+  def up
+    ScriptLevel.where.not(properties: nil).each do |script_level|
+      obj = JSON.parse script_level.properties
+      script_level.properties = { variants: obj }.to_json
+      script_level.save
```

Comment:
http://apidock.com/rails/v4.2.7/ActiveRecord/Persistence/save
http://apidock.com/rails/v4.2.7/ActiveRecord/Persistence/update%21

Mostly, whether it raises an exception. The other is that `update` or `update!` both modifying values and saves, whereas `save` or `save!` requires the modifying values be done elsewhere.

Not sure what do with exceptions. Maybe simply log them. But silently dropping them (which `save` does) doesn't seem appropriate.